x-pack/plugins/observability_ai_assistant/public/components/prompt_editor/prompt_editor.tsx

```diff
@@ -113,9 +124,11 @@ export function ChatPromptEditor({
   // Submit on Enter
   useEffect(() => {
     const keyboardListener = (event: KeyboardEvent) => {
-      if (!event.shiftKey && event.key === keys.ENTER && innerMessage) {
-        event.preventDefault();
-        handleSubmit();
+      if (innerMessage && !disabled && !invalid && hasFocus) {
+        if (!event.shiftKey && event.key === keys.ENTER) {
+          event.preventDefault();
+          handleSubmit();
+        }
```

Comment:
This is a complicated if, can we simplify? (I don't think we need two `if`s anyway?

Reply:
Yeah, the idea was future proofing, allowing easier additional other keyboard commands. First `if` checking if it should continue, send the key combination. But if we think no additional kb commands will be added I can consolidate.